dips/dip-182.md

```diff
@@ -0,0 +1,71 @@
+---
+dip: 182
+title: On-Chain Domain Resolution
+authors: Sunmi Lee (@sunmilee), David Wolinsky (@davidiw)
+status: Draft
+type: Informational
+created: 06/08/2021
+last_updated: 06/015/2021
+issue: https://github.com/diem/dip/issues/182
+---
+
+# Summary
+
+This DIP describes the on-chain domain name resolution process.
+
+On-chain domain name resolution allows organizations to be able to define their product names, and for VASPs to look up other VASPs using the product names.
+Having a human-readable VASP identifier gives Diem Ecosystem the ability enumerate all VASPs by their names, which is useful for merchant purposes as well as p2p transaction.
+For example, given a VASP domain name in a payment transaction, VASPs can look up where to route the payment correctly using an approved and accurate mapping of VASP identifier to an on-chain address.
+
+# On-chain data
+
+## Domain Format
+A VASP domain name is a unique string that is mapped to a VASP. Specification:
+* Case insensitive
+* Valid regular expression: `^[a-zA-Z0-9][a-zA-Z0-9.]*$`
+* Maximum length: 63 characters (64 including `@`)
+
+## DiemID Domain
+
+```
+resource struct DiemIdDomains {
+    domains: vector<DiemIdDomain>,
+}
+struct DiemIdDomain {
+    domain: vector<u8>,  // Utf-8 encoded
+}
+```
+* Field definitions:
+    * `domain` - name of a DiemID domain
+* The `DiemIdDomains` resource can only be published into an account with `Role == PARENT_VASP_ROLE_ID`.
+* The `DiemIdDomains` contains a list of `DiemIdDomain` structs that are associated with a VASP. As such, it is possible to register more than one DiemID Domain for a given VASP
+* Only the special TreasuryCompliance account (address `0xB1E55ED`) can manipulate a DiemIdDomains resource:
+    * Only TC account can create and publish a `DiemIdDomains` resource
+    * Only TC account can add, remove or update a `DiemIdDomain` within `DiemIdDomains` resource
+* `DiemIDDomains` resource will be created in an empty state as part of creating a `ParentVASP` account resource, and existing `ParentVASP` accounts without `DiemIDDomains` will have the resource instantiated by the DiemRoot account
+* In order to register a DiemID domain, a VASP needs to submit a request to Diem Association. After approval, Diem Association will use TreasuryCompliance account to issue an on-chain transaction to register a DiemID Domain
```

Comment:
Is the Association responsible for ensuring the uniqueness of all VASP domain names at any given time?  Across all history?  Will domain name re-used be supported, say after a delay, similar to ICANN?

Reply:
I don't think this is the appropriate doc to describe processes like that.

Reply:
Where will we have that information?  If folks are to use this feature they will need to know these types of things.

Reply:
Fair to some of your points. I agree it is relevant to say that all names are to be unique and that is the responsibility of the TC maintainers to ensure uniqueness.